source/vision/providerBase.py

```diff
@@ -7,35 +7,130 @@
 """Module within the vision framework that contains the base vision enhancement provider class.
 """
 
-import driverHandler
-from abc import abstractmethod
-from .constants import Role
+from abc import abstractmethod, ABC
+
+from autoSettingsUtils.autoSettings import AutoSettings
+from baseObject import AutoPropertyObject
 from .visionHandlerExtensionPoints import EventExtensionPoints
-from typing import FrozenSet
+from typing import Optional, Any
 
 
-class VisionEnhancementProvider(driverHandler.Driver):
-	"""A class for vision enhancement providers.
+class VisionEnhancementProviderSettings(AutoSettings, ABC):

```

Comment:
I believe that AutoSettings is already using ABC, as it inherrits from AutoPropertyObject.

```suggestion
class VisionEnhancementProviderSettings(AutoSettings):
```